unlock-js/src/unlockService.js

```diff
@@ -136,4 +151,134 @@ export default class UnlockService extends EventEmitter {
     }
     return version
   }
+
+  /**
+   * Returns the implementation based on the deployed version
+   * @param {*} address
+   */
+  async ethers_unlockContractAbiVersion() {
+    if (!this.provider) {
+      throw new Error(Errors.MISSING_WEB3)
+    }
+
+    let version = this.ethers_versionForAddress[this.unlockContractAddress]
+    if (version === undefined) {
+      // This was not memo-ized
+      version = await this._ethers_getVersionFromContract(
+        this.unlockContractAddress
+      )
+      this.ethers_versionForAddress[this.unlockContractAddress] = version
+    }
+
+    if (1 === version) {
+      return v01
+    }
+
+    if (2 === version) {
+      return v02
+    }
+
+    // Defaults to v0
+    return v0
+  }
+
+  /**
+   * Returns the ABI for the Lock contract deployed at the provided address
+   * @param {*} address
+   */
+  async ethers_lockContractAbiVersion(address) {
+    if (!this.provider) {
+      throw new Error(Errors.MISSING_WEB3)
+    }
+
+    let version = this.ethers_versionForAddress[address.toLowerCase()]
+    if (version === undefined) {
+      // This was not memo-ized
+      version = await this._ethers_getPublicLockVersionFromContract(address)
+      this.ethers_versionForAddress[address.toLowerCase()] = version
```

Comment:
Same as above questions about the lowercasing. Would it be a good idea to have a setter for `ethers_versionForAddress` method that handles conversion to lowercase?

Reply:
since it is now only used in 2 places, I think I'll just use a local variable, but if this becomes a larger question, we could use the checksummed address as our default